core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -136,6 +130,21 @@ private DBWriter() {
         });
     }
 
+    /**
+     * Deletes a downloaded FeedMedia file from the storage device.
+     *
+     * @param context A context that is used for opening a database connection.
+     * @param mediaId ID of the FeedMedia object whose downloaded file should be deleted.
+     */
+    public static Future<?> deleteFeedMediaOfItem(final Context context,
+                                                  final long mediaId) {
+        Future res = doDeleteFeedMediaOfItem(context, mediaId);
+        if (UserPreferences.shouldDeleteRemoveFromQueue()) {
+            DBWriter.removeQueueItem(context, DBReader.getFeedMedia(mediaId).getItem(), false);
+        }
+        return res;
```

Comment:
Then instead of returning the `Future` of `doDeleteFeedMediaOfItem()`, return the `Future` of `DBWriter.removeQueueItem()`.